python/lsst/ts/externalscripts/base_make_calibrations.py

```diff
@@ -0,0 +1,640 @@
+        )
+
+        self.config = config
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script.
+        """
+        metadata.duration = self.config.n_bias*(self.camera.read_out_time + self.estimated_process_time)
+
+    async def take_image_type(self, image_type, exp_times):
+        """Take exposures and build exposure set.
+
+        Parameters
+        ----------
+        image_type : `str`
+            Image type. One of ["BIAS", "DARK", "FLAT"].
+
+        exp_times: `list`
+            List of exposure times.
+
+        Returns
+        -------
+            Tuple with exposure IDs.
+        """
+
+        return tuple(
+            [
+                (await self.camera.take_imgtype(image_type, exp_time, 1))[0]
+                for exp_time in exp_times
+            ]
+        )
+
+    async def image_in_oods_callback(self, data):
+        """Callback function to check images are in archiver"""
+        self.image_in_oods_samples[self.current_image_type].append(data)
+        self.number_of_images_taken += 1
+        if self.number_of_images_taken == self.number_of_images_expected:
+            self.image_in_oods_received_all_expected.set()
+
+    async def take_images(self, image_type):
+        """Take images with instrument.
+
+        Parameters
+        ----------
+        image_type : `str`
+            Image type. One of ["BIAS", "DARK", "FLAT"].
+
+        Returns
+        -------
+        exposures : `tuple`
+             Tuple with the IDs of the exposures taken.
+        """
+
+        exp_times = await self.set_exp_times_per_im_type(image_type)
+
+        n_detectors = len(tuple(map(int, self.config.detectors[1:-1].split(','))))
+
+        self.number_of_images_expected = len(exp_times)*n_detectors
+        self.number_of_images_taken = 0
+        self.image_in_oods_received_all_expected.clear()
+        self.current_image_type = image_type
+
+        # callback
+        self.image_in_oods.callback = self.image_in_oods_callback
+
+        exposures = await self.take_image_type(image_type, exp_times)
+
+        await asyncio.wait_for(self.image_in_oods_received_all_expected.wait(),
+                               timeout=self.config.oods_timeout)
+
+        self.ocps.evt_job_result.flush()
+
+        return exposures
+
+    async def call_pipetask(self, image_type, exposure_ids_dict):
+        """Call pipetasks via the OCPS.
+
+        Parameters
+        ----------
+        image_type : `str`
+            Image or calibration type. One of ["BIAS", "DARK",
+            "FLAT", "DEFETCS", "PTC"].
+
+        exposure_ids_dict: `dict` [`str`]
+            Dictionary with tuple with exposure IDs for "BIAS",
+            "DARK", or "FLAT".
+
+        Returns
+        -------
+        response : `dict`
+             Dictionary with the final OCPS status.
+        """
+
+        # Run the pipetask via the OCPS
+        if image_type == "BIAS":
+            pipe_yaml = "cpBias.yaml"
```

Comment:
This is where my comment above about camera specific pipelines would apply.  I think the names can be constructed from the camera's name, but I'm not 100% sure.

Reply:
I had to do some if/else clauses using the instrument name method, because of different capitalizations in the path names and the instrument names that the pipetask takes (e.g., LATSS vs Latiss and LSSTcomCam vs LsstComCam).